subgraphs/openzeppelin-governor/protocols/silo-governance/src/silo-governor.ts

```diff
@@ -0,0 +1,129 @@
+export function handleProposalCreated(event: ProposalCreated): void {
+  // FIXME: Prefer to use a single object arg for params
+  // e.g.  { proposalId: event.params.proposalId, proposer: event.params.proposer, ...}
+  // but graph wasm compilation breaks for unknown reasons
+
+  _handleProposalCreated(
+    event.params.proposalId.toString(),
+    event.params.proposer.toHexString(),
+    event.params.targets,
+    event.params.values,
+    event.params.signatures,
+    event.params.calldatas,
+    event.params.startBlock,
+    event.params.endBlock,
+    event.params.description,
+    event
+  );
+}
+
+// ProposalExecuted(proposalId)
+export function handleProposalExecuted(event: ProposalExecuted): void {
+  _handleProposalExecuted(event.params.proposalId.toString(), event);
+}
+
+// ProposalQueued(proposalId, eta)
+export function handleProposalQueued(event: ProposalQueued): void {
+  _handleProposalQueued(event.params.proposalId, event.params.eta);
+}
+
+export function handleProposalThresholdSet(event: ProposalThresholdSet): void {
+  let governanceFramework = getGovernanceFramework(event.address.toHexString());
+  governanceFramework.proposalThreshold = event.params.newProposalThreshold;
+  governanceFramework.save();
+}
+
+// QuorumNumeratorUpdated(oldQuorumNumerator, newQuorumNumerator)
+export function handleQuorumNumeratorUpdated(
+  event: QuorumNumeratorUpdated
+): void {
+  let governanceFramework = getGovernanceFramework(event.address.toHexString());
+  governanceFramework.quorumNumerator = event.params.newQuorumNumerator;
+  governanceFramework.save();
+}
+
+// TimelockChange (address oldTimelock, address newTimelock)
+export function handleTimelockChange(event: TimelockChange): void {
+  let governanceFramework = getGovernanceFramework(event.address.toHexString());
+  governanceFramework.timelockAddress = event.params.newTimelock.toHexString();
+  governanceFramework.save();
+}
+
+// VoteCast(account, proposalId, support, weight, reason);
+export function handleVoteCast(event: VoteCast): void {
+  _handleVoteCast(
+    event.params.proposalId.toString(),
+    event.params.voter.toHexString(),
+    event.params.weight,
+    event.params.reason,
+    event.params.support,
+    event
+  );
+}
+
+export function handleVotingDelaySet(event: VotingDelaySet): void {
+  let governanceFramework = getGovernanceFramework(event.address.toHexString());
+  governanceFramework.votingDelay = event.params.newVotingDelay;
+  governanceFramework.save();
+}
+
+export function handleVotingPeriodSet(event: VotingPeriodSet): void {
+  let governanceFramework = getGovernanceFramework(event.address.toHexString());
+  governanceFramework.votingPeriod = event.params.newVotingPeriod;
+  governanceFramework.save();
+}
+
+// Helper function that imports and binds the contract
+function getGovernanceFramework(contractAddress: string): GovernanceFramework {
```

Comment:
I wonder if we could make this a generic function and pass down the contract to bind to as T 🤔

Reply:
thats a good idea, let me check if out

Reply:
i remembered the issue, if i abstract it away it would no longer catch mistakes with diff config
![image](https://user-images.githubusercontent.com/4507317/179251689-fcfc2218-527b-42de-99f9-7c08b3e1d44e.png)